File: libs/contracts/DeviceState.cs
namespace Atalaya.Contracts;

/// <summary>
/// Read model del camino caliente (SAD §6, ADR-005): última posición/estado conocido
/// por dispositivo. Una instancia por dispositivo, actualizada por el procesamiento y
/// servida al dashboard como snapshot inicial y como delta en vivo por SignalR.
/// </summary>
public sealed record DeviceState(
    string DeviceId,
    DateTimeOffset Ts,
    long Seq,
    double Lat,
    double Lng,
    double SpeedKmh,
    double HeadingDeg,
    double FuelPct,
    double EngineTempC)
{
    public static DeviceState FromEvent(TelemetryEvent e) => new(
        e.DeviceId, e.Ts, e.Seq, e.Lat, e.Lng,
        e.SpeedKmh, e.HeadingDeg, e.FuelPct, e.EngineTempC);
}

File: apps/atalaya-web/src/app/core/models/device-state.ts
/** Read model del camino caliente (espejo de Atalaya.Contracts.DeviceState). */
export interface DeviceState {
  deviceId: string;
  ts: string;
  seq: number;
  lat: number;
  lng: number;
  speedKmh: number;
  headingDeg: number;
  fuelPct: number;
  engineTempC: number;
}
